WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/post/PostCardBuilder.kt

```diff
@@ -1,35 +1,47 @@
 package org.wordpress.android.ui.mysite.cards.post
 
+import org.wordpress.android.R
 import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardDraftOrScheduled
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.PostCard.PostCardDraftOrScheduled.PostItem
 import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.PostCardBuilderParams
+import org.wordpress.android.ui.mysite.cards.post.mockdata.MockedPostsData.Post
+import org.wordpress.android.ui.utils.UiString.UiStringRes
 import org.wordpress.android.ui.utils.UiString.UiStringText
 import javax.inject.Inject
 
 class PostCardBuilder @Inject constructor() {
-    fun build(params: PostCardBuilderParams): List<PostCard> {
-        val cards = mutableListOf<PostCard>()
-        params.mockedPostsData?.posts?.draft?.map {
-            cards.add(
-                    PostCard(
-                            title = UiStringText(DRAFT_TITLE),
-                            postTitle = UiStringText(it.title ?: NO_TITLE)
-                    )
-            )
-        }
-        params.mockedPostsData?.posts?.scheduled?.map {
-            cards.add(
-                    PostCard(
-                            title = UiStringText(SCHEDULED_TITLE),
-                            postTitle = UiStringText(it.title ?: NO_TITLE)
-                    )
-            )
-        }
-        return cards
+    fun build(params: PostCardBuilderParams) = mutableListOf<PostCard>().apply {
+        val posts = params.mockedPostsData?.posts
+        posts?.draft?.takeIf { it.isNotEmpty() }?.let { add(it.createDraftPostsCard()) }
+        posts?.scheduled?.takeIf { it.isNotEmpty() }?.let { add(it.createScheduledPostsCard()) }
     }
 
-    companion object {
-        const val DRAFT_TITLE = "Draft Post"
-        const val SCHEDULED_TITLE = "Scheduled Post"
-        const val NO_TITLE = "No title"
+    private fun List<Post>.createDraftPostsCard() = PostCardDraftOrScheduled(
+            postCardType = PostCardType.DRAFT,
+            title = UiStringRes(R.string.my_site_post_card_draft_title),
+            postItems = mapToDraftOrScheduledPostItems(PostCardType.DRAFT)
+    )
+
+    private fun List<Post>.createScheduledPostsCard() = PostCardDraftOrScheduled(
+            postCardType = PostCardType.SCHEDULED,
+            title = UiStringRes(R.string.my_site_post_card_scheduled_title),
+            postItems = mapToDraftOrScheduledPostItems(PostCardType.SCHEDULED)
+    )
+
+    private fun List<Post>.mapToDraftOrScheduledPostItems(postCardType: PostCardType) = map { post ->
+        val excerpt = if (postCardType == PostCardType.SCHEDULED) {
+            "Today at 1:04 PM" // TODO: ashiagr - remove hardcoded text
```

Comment:
Question (❓): This will be updated in a separate PR, right?

Reply:
Right, it will be updated in a separate PR based on the modified date format in the API.

Reply:
Great, thanks for letting me know Ashita! 🙏